core/src/main/java/com/linecorp/armeria/common/util/AsyncCloseableSupport.java

```diff
@@ -0,0 +1,203 @@
+ */
+public final class AsyncCloseableSupport implements AsyncCloseable {
+
+    private static final AtomicIntegerFieldUpdater<AsyncCloseableSupport> closingUpdater =
+            AtomicIntegerFieldUpdater.newUpdater(AsyncCloseableSupport.class, "closing");
+
+    /**
+     * Returns a new {@link AsyncCloseableSupport} that will be completed immediately on {@link #close()} or
+     * {@link #closeAsync()}. This method is useful when you don't have any resources to release.
+     * <pre>{@code
+     * > public class MyClass {
+     * >     final AsyncCloseableSupport closeableSupport = AsyncCloseableSupport.of();
+     * >
+     * >     public void doSomething() {
+     * >         if (closeableSupport.isClosing()) {
+     * >             throw new IllegalStateException("Closed already");
+     * >         }
+     * >         ...
+     * >     }
+     * >
+     * >     @Override
+     * >     public CompletableFuture<?> closeFuture() {
+     * >         return closeableSupport.closeFuture();
+     * >     }
+     * >
+     * >     @Override
+     * >     public CompletableFuture<?> closeAsync() {
+     * >         return closeableSupport.closeAsync();
+     * >     }
+     * >
+     * >     @Override
+     * >     public CompletableFuture<?> close() {
+     * >         return closeableSupport.close();
+     * >     }
+     * > }
+     * }</pre>
+     */
+    public static AsyncCloseableSupport of() {
+        return of(f -> f.complete(null));
+    }
+
+    /**
+     * Returns a new {@link AsyncCloseableSupport} which calls the specified {@link Consumer} on
+     * {@link #close()} or {@link #closeAsync()}.
+     * <pre>{@code
+     * > class MyClass implements AutoCloseable {
+     * >     final AsyncCloseableSupport closeableSupport = AsyncCloseableSupport.of(f -> {
+     * >         // Release resources here.
+     * >         ...
+     * >         f.complete(null);
+     * >     });
+     * >
+     * >     @Override
+     * >     public CompletableFuture<?> closeFuture() {
+     * >         return closeableSupport.closeFuture();
+     * >     }
+     * >
+     * >     @Override
+     * >     public CompletableFuture<?> closeAsync() {
+     * >         return closeableSupport.closeAsync();
+     * >     }
+     * >
+     * >     @Override
+     * >     public CompletableFuture<?> close() {
+     * >         return closeableSupport.close();
+     * >     }
+     * > }
+     * }</pre>
+     *
+     * @param closeAction the {@link Consumer} which performs the task that release the resources and
+     *                    completes the given {@link CompletableFuture}.
+     */
+    public static AsyncCloseableSupport of(Consumer<CompletableFuture<?>> closeAction) {
+        return new AsyncCloseableSupport(requireNonNull(closeAction, "closeAction"));
+    }
+
+    /**
+     * Returns the {@link AsyncCloseableSupport} which has been closed already.
+     */
+    public static AsyncCloseableSupport closed() {
```

Comment:
Not sure if we need this method, but if we do, how about making a static variable for this?